forem/forem
# Dynamic frontend for the sidebar navigation Items

## What type of PR is this? (check all applicable)


## Description

In this PR I've created a dynamic frontend that uses the new NavigationLinks table. This means that the sidebar is no longer hardcoded but can be configured from http://localhost:3000/admin/navigation_links. 

In addition to this PR there is also related docs which you can find https://app.gitbook.com/@forem/s/forem-admin-guide/~/drafts/-MJN9RF0nZtuN9YWy-IM/admin/navigation-links/@drafts

## Related Tickets & Documents
Follow up PR of  https://github.com/forem/forem/pull/9444 and https://github.com/forem/forem/pull/10662
https://github.com/forem/forem/issues/4661
https://github.com/forem/forem/issues/5475

## QA Instructions, Screenshots, Recordings

1. Open up a rails console
2. Run `NavigationLink.all`, ensure that the count matches the sidebar item count on http://localhost:3000/
3. Add, edit and delete items and ensure that it shows the updated changes on the homepage.

A view of the backend:
<img width="1193" alt="Screenshot 2020-10-12 at 11 46 38" src="https://user-images.githubusercontent.com/2786819/95732088-a34cd200-0c80-11eb-9a09-b31efa2f627c.png">

A view of the corresponding frontend:
<img width="294" alt="Screenshot 2020-10-12 at 11 46 22" src="https://user-images.githubusercontent.com/2786819/95732095-a5af2c00-0c80-11eb-841d-bcf4f9e1d61a.png">

## Added tests?


## Added to documentation?


## [optional] Are there any post deployment tasks we need to perform?

## [optional] What gif best describes this PR or how it makes you feel?

![alt_text](https://media.giphy.com/media/TIdToBm873KBoSkPlb/giphy.gif)
